src/common/copilot/user-feedback/CESSurvey.ts

```diff
@@ -0,0 +1,148 @@
+          feedbackPanel.dispose();
+          break;
+      }
+    },
+    undefined,
+    context.subscriptions
+  );
+}
+
+function createFeedbackPanel(context: vscode.ExtensionContext): vscode.WebviewPanel {
+  const feedbackPanel = vscode.window.createWebviewPanel(
+    "CESUserFeedback",
+    "Feedback",
+    vscode.ViewColumn.Seven,
+    {
+      enableScripts: true,
+    }
+  );
+
+  context.subscriptions.push(feedbackPanel);
+
+  return feedbackPanel;
+}
+
+function getWebviewURIs(context: vscode.ExtensionContext, feedbackPanel: vscode.WebviewPanel): { feedbackCssUri: vscode.Uri, feedbackJsUri: vscode.Uri } {
+  const feedbackCssPath = vscode.Uri.joinPath(context.extensionUri, 'src', 'common', 'copilot', "user-feedback", "feedback.css");
+  const feedbackCssUri = feedbackPanel.webview.asWebviewUri(feedbackCssPath);
+
+  const feedbackJsPath = vscode.Uri.joinPath(context.extensionUri, 'src', 'common', 'copilot', "user-feedback", "feedback.js");
+  const feedbackJsUri = feedbackPanel.webview.asWebviewUri(feedbackJsPath);
+
+  return { feedbackCssUri, feedbackJsUri };
+}
+
+function initializeFeedbackData(sessionId: string): FeedbackData {
+  const feedbackData: FeedbackData = {
+    IsDismissed: false,
+    ProductContext: [
+      {
+        key: 'sessionId',
+        value: sessionId
+      },
+      {
+        key: 'scenario',
+        value: 'ProdDevCopilot'
+      }
+    ],
+    Feedbacks: [
+      {
+        key: 'comment',
+        value: ''
+      }
+    ]
+  };
+
+  return feedbackData;
+}
+
+async function handleFeedbackSubmission(text: string, endpointUrl: string, apiToken: string, feedbackData: FeedbackData, telemetry: ITelemetry) {
+  feedbackData.Feedbacks[0].value = text;
+
+  try {
+    const response = await fetch(endpointUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + apiToken,
+      },
+      body: JSON.stringify(feedbackData)
+    });
+
+    if (response.ok) {
+      // Feedback sent successfully
+      const responseJson = await response.json();
+      const feedbackId = responseJson.FeedbackId;
+      sendTelemetryEvent(telemetry, { eventName: UserFeedbackSuccessEvent, FeedbackId: feedbackId });
+    } else {
+      // Error sending feedback
+      sendTelemetryEvent(telemetry, { eventName: UserFeedbackFailureEvent, error: response.statusText });
+    }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    // Network error or other exception
+    sendTelemetryEvent(telemetry, { eventName: UserFeedbackFailureEvent, exception: error.message });
+  }
+}
+
+function getWebviewContent(feedbackCssUri: vscode.Uri, feedbackJsUri: vscode.Uri, nonce: string) {
+
+  return `<!DOCTYPE html>
+    <html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
```

Comment:
What about CSP? Why it is not integrated?
@ashishchoudhary001 FYI

Reply:
Will add it in the next PR